tools/flang1/flang1exe/ast.c

```diff
@@ -918,8 +918,13 @@ mk_binop(int optype, int lop, int rop, DTYPE dtype)
           INT v;
           v1 = CONVAL2G(A_SPTRG(lop));
           v2 = CONVAL2G(A_SPTRG(rop));
-          if (v2 < 0)
+          if (v2 < 0) {
+            if (v1 == -1 && (v2 & 1))
```

Comment:
Can you add a comment here saying if v2 is odd then the result will be -1. Can you also add a test with an odd number which is not -1?

Reply:
The comment has been added in the new commit. Testcase of odd base with exponent which is not -1 has also been added.